appservice/src/intTest/groovy/nu/yona/server/DisclosureTest.groovy

```diff
@@ -0,0 +1,358 @@
+			{
+				disclosureRequest1URL = response.responseData._embedded.goalConflictMessages[0]._links.requestDisclosure.href
+				disclosureRequest2URL = response.responseData._embedded.goalConflictMessages[1]._links.requestDisclosure.href
+			}
+
+		then:
+			response.status == 200
+			response.responseData._embedded.goalConflictMessages.size() == 2
+			response.responseData._embedded.goalConflictMessages[0].nickname == "RQ ${timestamp}"
+			response.responseData._embedded.goalConflictMessages[0].goalName == "news"
+			response.responseData._embedded.goalConflictMessages[0].url == null
+			response.responseData._embedded.goalConflictMessages[0]._links.requestDisclosure
+			response.responseData._embedded.goalConflictMessages[1].nickname == "RQ ${timestamp}"
+			response.responseData._embedded.goalConflictMessages[1].goalName == "gambling"
+			response.responseData._embedded.goalConflictMessages[1].url == null
+			response.responseData._embedded.goalConflictMessages[1]._links.requestDisclosure
+	}
+	
+	def 'Richard does not have disclosure request links on his own goal conflict messages'(){
+		given:
+
+		when:
+			def response = appService.getAnonymousMessages(richardQuinURL, richardQuinPassword)
+
+		then:
+			response.status == 200
+			response.responseData._embedded.goalConflictMessages.size() == 2
+			response.responseData._embedded.goalConflictMessages[0].nickname == "<self>"
+			response.responseData._embedded.goalConflictMessages[0].goalName == "news"
+			response.responseData._embedded.goalConflictMessages[0].url == "http://www.refdag.nl"
+			!response.responseData._embedded.goalConflictMessages[0]._links.requestDisclosure
+			response.responseData._embedded.goalConflictMessages[1].nickname == "<self>"
+			response.responseData._embedded.goalConflictMessages[1].goalName == "gambling"
+			response.responseData._embedded.goalConflictMessages[1].url == "http://www.poker.com"
+			!response.responseData._embedded.goalConflictMessages[1]._links.requestDisclosure
+	}
+
+	def 'Bob asks for disclosure of both'(){
+		given:
+
+		when:
+			def response1 = appService.postMessageActionWithPassword(disclosureRequest1URL, """{
+					"properties":{
+					}
+				}""", bobDunnPassword)
+			def response2 = appService.postMessageActionWithPassword(disclosureRequest2URL, """{
+					"properties":{
+					}
+				}""", bobDunnPassword)
+
+		then:
+			response1.status == 200
+			response2.status == 200
+	}
+
+	def 'Richard checks his anonymous messages and finds the disclosure request'(){
+		given:
+
+		when:
+			def response = appService.getAnonymousMessages(richardQuinURL, richardQuinPassword)
+			if(response.status == 200 && response.responseData._embedded.discloseRequestMessages) {
+				goalDiscloseRequestMessage1AcceptURL = response.responseData._embedded.discloseRequestMessages[0]._links.accept.href
+				goalDiscloseRequestMessage2RejectURL = response.responseData._embedded.discloseRequestMessages[1]._links.reject.href
+			}
```

Comment:
The request should contain a link to the message for which the disclose is requested.